setup.py

```diff
@@ -0,0 +1,39 @@
+from setuptools import setup
+
+test_dependencies = [
+    "coverage",
+    "pytest",
+    "pytest-cov",
+]
+
+extras = {
+    "testing": test_dependencies,
+}
+
+with open("README.md", "rt") as f:
+    long_description = f.read()
+
+setup(
```

Comment:
I'm not sure if using `setup.py` is the right thing for this project of if we should instead just use a `requirements.txt`. My feeling here is that since we'll only use it in airflow and it doesn't need to be installed by other users, going with `requirements.txt` is the right way here. I found a good explanation here: https://caremad.io/posts/2013/07/setup-vs-requirement/

But I might be wrong.

Reply:
You're not wrong. I think dstufft's main point was "please stop pinning dependencies in setup.py if it's going on PyPI; that's for abstract dependencies."

Some things that come with a setup.py include having a notion of a build process and a process for installation. I think I agree we don't really need these. They're kind of nice to have because e.g. the `tox` setup is more familiar and they're "standard" but we won't really leverage them.

I think the most useful thing about the setup.py is that it contains a list of abstract dependencies that we can produce a frozen list of concrete dependencies from at will. A frozen environment a la requirements.txt will contain the recursive dependencies of our dependencies and updating that by hand when we want to upgrade one of our direct deps is frustrating.

So I think it's useful to have both.